### PR TITLE
test: re-enable window function over parquet with forced collisions

### LINKED ISSUE #11660: Window function test fails when `force_hash_collisions` is enabled
### Describe the bug

The `force_hash_collisions` feature flag is intended to ensure DataFusion works correctly even in the very unlikely event that hash values collide.

The new version of clippy in Rust 1.80 found we had not been running some tests with this flag  https://github.com/apache/datafusion/issues/11651

And it turns out when we started running the tests with the hash collision they failed

we commented them out in https://github.com/apache/datafusion/pull/11654 but we should really figure out why they failed

Maybe the test needs to be updated or maybe there is a real bug

```shell
cargo test --test sqllogictests --features=force_hash_collisions -- parquet
```

### To Reproduce

Uncomment the tests in `parquet.slt` that was commented out in https://github.com/apache/datafusion/pull/11654 and then run

```shell
cargo test --test sqllogictests --features=force_hash_collisions -- parquet
```

### Expected behavior

Tests should pass

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by korowa**
## Which issue does this PR close?

Closes #11660.

## Rationale for this change

After several attempts to reproduce the failure locally and in fork CI it still was not reproducing. Also I wasn't able to find any workflow failure related to 1.80 upgrade in PRs and on main. So, trying to just uncomment the test.

## What changes are included in this PR?

Test uncommented.

### CODE REVIEW

**Review by alamb — APPROVED** [review]
Thank you @korowa 

FYI @nrc who originally reported this was failing

**Comment by alamb** [comment]
I re-ran the relevant test: 

```shell
cargo test --test sqllogictests --features=force_hash_collisions -- parquet
```

And indeed it passes locally for me too

```
andrewlamb@Andrews-MBP-2:~/Software/datafusion$ cargo test --test sqllogictests --features=force_hash_collisions -- parquet
warning: function `hash_map_array` is never used
   --> datafusion/common/src/hash_utils.rs:249:4
    |
249 | fn hash_map_array(
    |    ^^^^^^^^^^^^^^
    |
    = note: `#[warn(dead_code)]` on by default

warning: `datafusion-common` (lib) generated 1 warning
    Finished `test` profile [unoptimized + debuginfo] target(s) in 0.40s
     Running bin/sqllogictests.rs (target/debug/deps/sqllogictests-a3320b7543efea67)
Running "parquet.slt"
Running "parquet_sorted_statistics.slt"
andrewlamb@Andrews-MBP-2:~/Software/datafusion$
```

File: datafusion/sqllogictest/test_files/parquet.slt [hunk 1]
```diff
@@ -251,27 +251,25 @@ SELECT COUNT(*) FROM timestamp_with_tz;
 ----
 131072
 
-# FIXME(#TODO) fails with feature `force_hash_collisions`
-# https://github.com/apache/datafusion/issues/11660
 # Perform the query:
```
Comment (comphead) [inline datafusion/sqllogictest/test_files/parquet.slt:254]:
```suggestion
```

Comment (comphead) [reply datafusion/sqllogictest/test_files/parquet.slt:254]: Would be great to add some comments why test is so specific

Comment (alamb) [reply datafusion/sqllogictest/test_files/parquet.slt:254]: I agree better comments would be nice -- however I don't think this PR makes the comment any more/less cryptic, so I don't think we should hold it for better comments.

Maybe we can improve the comments as a future PR

**Review by alamb — APPROVED** [review]
Thanks agian @korowa  and @comphead

**Pull request merged**